### ISSUE TITLE
Invalid_argument on Windows

### ISSUE THREAD

**Issue by msprotz** [the-lambda-church/merlin#594]
Fatal error: exception Sys_error("Invalid argument")
Called from file "src/frontend/ocamlmerlin.ml", line 145, characters 6-95
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/utils/std.ml", line 672, characters 16-20
Re-raised at file "src/utils/std.ml", line 675, characters 20-23
Called from file "src/frontend/ocamlmerlin.ml", line 166, characters 2-14

Let me know what I can do to help

**Comment by raphael-proust** [comment]
Is that at start-up? When executing a specific command?

Also, can you post version information for merlin, ocaml and whatever editor you are invoking merlin from?

**Comment by msprotz** [comment]
When executing `gd` in vim. (Go to definition.)

```
protz@Joprotze-Z420:~ $ ocamlmerlin -version
The Merlin toolkit version git-6480e585a0e9d036d11aaf28bcee97e8e9b77c2e, for Ocaml 4.02.3
protz@Joprotze-Z420:~ $ ocaml -version
The OCaml toplevel, version 4.02.3
```

The editor is gvim, a cygwin program, executing ocamlmerlin, a native windows program.

**Comment by let-def** [comment]
@msprotz: I will try to reproduce this problem. Can you give information on your windows setup?
(I guess this is your build of ocaml for windows? And how can I get the gvim binary? Sorry I am not familiar with this environment)

**Comment by msprotz** [comment]
Yes. The gvim binary is the one from cygwin -- all of my packages are from Cygwin, for consistency, but then the merlin and ocaml binaries are native windows binaries, which makes things kind of weird.

You should be able to install vim (or gvim, but I suspect it doesn't make much of a difference) with cygwin's setup.exe

**Comment by let-def** [comment]
I have been trying cygwin in the last 4 days, it failed everytime, even on a fresh VM.
Seems they are (they were at least this week-end) updating packages, and synchronization fails everytime.
I have rarely seen an installer failing so unreliably :P.

**Comment by msprotz** [comment]
Urgh. Things are so very broken.

Wondering whether @dra27's work would help here.

**Comment by dra27** [comment]
@let-def - have you tried disabling all virus scanners (including Windows Defender). It's usually the underlying cause of random failures.
@msprotz - really must get that Cygwin process invocation into a wider library, yes!
